### DISCUSSION TITLE
Project Status?

### DISCUSSION THREAD

**Discussion by subtosharki** [pintariching/rustle#70] [General]
I just came across this project on this [reddit thread](https://www.reddit.com/r/rust/comments/dc5s7b/are_there_any_svelteinspired_front_end_frameworks/) and saw that six months ago for some reason all of the code was removed and titled "restart from scratch" 

im curious if this is being worked on in the background as i would love to use this to convert my codebase on a project to rust to i can take advantages of workspaces

**Comment by pintariching** [comment]
I'm currently not actively developing this, feel free to fork the project or if you want, I can transfer the project to you.

**Reply by subtosharki** [reply]
i have no experience writing a compiler but have always been interested, any leads on how to get started? i would love to make this a thing, especially if i cant get something like the example below working somehow and maintain the speed of svelte.
```
<script lang="rs">
    let foo = 5;
    $: bar = foo * 2;
 
    let handle_click = || {
         foo += 1;
    }
</script>

<button on:click={handle_click}>
    count: {foo}
</button>

<h1>{foo} * 2 = {bar}</h1>

<p>{|| {
    format!("Value from a closure: {}", foo)
}}</p>
```
 i was looking at other frameworks and none give the experience svelte does

**Reply by pintariching** [reply]
The way I started was by watching lihautan on youtube, I think he is a developer on svelte. [I started with this video](https://youtu.be/mwvyKGw2CzU?si=njrtCPcp2sxMF8pS) and followed along in rust. Now instead of using a handwritten parser I tried to use `nom` I think to parse the svelte code into an AST. Then from that you generate the javascript file.

**Reply by subtosharki** [reply]
thanks! in your last discussion you mentioned it wouldnt be as fast as svelte, can you explain why? how could i optimize this?

in your last post you also mentioned you built to javascript, isnt the whole concept to be building to wasm?

**Reply by pintariching** [reply]
First I built it to compile the same javascript as svelte, the I tried to write wasm instead of js but I didn't get far. It would be slower because dom manipulation is slower in wasm than from js I think so there's no way around that unless they fixed it?

**Reply by subtosharki** [reply]
i appreciate the fast responses, im not a big frontend guy myself so i need to touch up on the DOM, i thought wasm would be faster than js overall though? its much closer to machine code and takes advantage of hardware much more than js, i understand if there is some js needed to glue it together (maybe not if they support it in wasm idk)

**Reply by pintariching** [reply]
It is much faster than js when doing complex calculations and multithreading but I'm not sure about DOM manipulation. I could be wrong though.

**Reply by subtosharki** [reply]
understood, its gonna be my first compiler so it will be very unstable probably, and it wont be my main project either. 

if you dont mine me asking, what made you give up maintaining the project?

**Reply by pintariching** [reply]
Honestly, it's just that my interest faded over time and I had other projects to work on. Good luck with your compiler, I hope to use it soon 😁

**Reply by subtosharki** [reply]
would love to see this maintained, like i said above i dont think i have enough rust skills to maintain it myself
